pkg/stanza/operator/input/udp/udp.go

```diff
@@ -150,42 +172,52 @@ func (u *Input) Start(_ operator.Persister) error {
 func (u *Input) goHandleMessages(ctx context.Context) {
 	u.wg.Add(1)
 
-	go func() {
-		defer u.wg.Done()
-
-		dec := decode.New(u.encoding)
-		buf := make([]byte, 0, MaxUDPSize)
-		for {
-			message, remoteAddr, err := u.readMessage()
-			if err != nil {
-				select {
-				case <-ctx.Done():
-					return
-				default:
-					u.Errorw("Failed reading messages", zap.Error(err))
-				}
-				break
-			}
+	numConcurrentReaders := 1
+	if u.AsyncConfig != (UdpAsyncConfig{}) {
+		numConcurrentReaders = u.AsyncConfig.FixedReaderRoutineCount
+	}
```

Comment:
Since `Build` handles the case where this value is not specified or is invalid, can we just assume it's valid here? e.g. `for i := 0; i < u.AsyncConfig.FixedReaderRoutineCount; i++ {`